include/nuttx/wireless/ieee80211/ieee80211.h

```diff
@@ -42,849 +42,4070 @@
+#define IEEE80211_HE_OPERATION_6GHZ_OP_INFO           0x00020000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_MASK         0x3f000000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_OFFSET       24
+#define IEEE80211_HE_OPERATION_PARTIAL_BSS_COLOR      0x40000000
+#define IEEE80211_HE_OPERATION_BSS_COLOR_DISABLED     0x80000000
+
+/* ieee80211_he_6ghz_oper - HE 6 GHz operation Information field
+ * @primary: primary channel
+ * @control: control flags
+ * @ccfs0: channel center frequency segment 0
+ * @ccfs1: channel center frequency segment 1
+ * @minrate: minimum rate (in 1 Mbps units)
  */
 
-#define ATHEROS_OUI                        ((const uint8_t[]){ 0x00, 0x03, 0x7f })
-#define BROADCOM_OUI                       ((const uint8_t[]){ 0x00, 0x90, 0x4c })
-#define IEEE80211_OUI                      ((const uint8_t[]){ 0x00, 0x0f, 0xac })
-#define MICROSOFT_OUI                     ((const uint8_t[]){ 0x00, 0x50, 0xf2 })
+struct ieee80211_he_6ghz_oper
+{
+  uint8_t primary;
+#define IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH         0x3
+#define  IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH_20MHZ  0
+#define  IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH_40MHZ  1
+#define  IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH_80MHZ  2
+#define  IEEE80211_HE_6GHZ_OPER_CTRL_CHANWIDTH_160MHZ 3
+#define IEEE80211_HE_6GHZ_OPER_CTRL_DUP_BEACON        0x4
+  uint8_t control;
+  uint8_t ccfs0;
+  uint8_t ccfs1;
+  uint8_t minrate;
+};
 
-#define IEEE80211_AUTH_ALGORITHM(auth) \
-    ((auth)[0] | ((auth)[1] << 8))
-#define IEEE80211_AUTH_TRANSACTION(auth) \
-    ((auth)[2] | ((auth)[3] << 8))
-#define IEEE80211_AUTH_STATUS(auth) \
-    ((auth)[4] | ((auth)[5] << 8))
+/* ieee80211_he_oper_size - calculate 802.11ax HE Operations IE size
+ * @he_oper_ie: byte data of the He Operations IE, stating from the byte
+ * after the ext ID byte. It is assumed that he_oper_ie has at least
+ * sizeof(struct ieee80211_he_operation) bytes, the caller must have
+ * validated this.
+ * @return the actual size of the IE data (not including header),
+ * or 0 on error
+ */
 
-/* Authentication Algorithm Number field (see 7.3.1.1). */
+static inline uint8_t
+ieee80211_he_oper_size(const uint8_t *he_oper_ie)
+{
+  struct ieee80211_he_operation *he_oper = (void *)he_oper_ie;
+  uint8_t oper_len = sizeof(struct ieee80211_he_operation);
+  uint32_t he_oper_params;
 
-#define IEEE80211_AUTH_ALG_OPEN            0x0000
-#define IEEE80211_AUTH_ALG_SHARED          0x0001
-#define IEEE80211_AUTH_ALG_LEAP            0x0080
+  /* Make sure the input is not NULL */
 
-/* WEP */
+  if (!he_oper_ie)
+    {
+      return 0;
+    }
 
-#define IEEE80211_WEP_KEYLEN               5   /* 40bit */
-#define IEEE80211_WEP_NKID                 4   /* number of key ids */
-#define IEEE80211_CHALLENGE_LEN            128
+  /* Calc required length */
 
-/* WEP header constants */
+  he_oper_params = le32_to_cpu(he_oper->he_oper_params);
+  if (he_oper_params & IEEE80211_HE_OPERATION_VHT_OPER_INFO)
```

Comment:
```suggestion
  if ((he_oper_params & IEEE80211_HE_OPERATION_VHT_OPER_INFO) != 0)
```

Reply:
How about keep the origin style here? it will easier to synchronize with upstream next time

Reply:
https://github.com/torvalds/linux/blob/master/include/linux/ieee80211.h

Reply:
So we are copying things from `SPDX-License-Identifier: GPL-2.0-only` and just put Apache 2.0 header on top? Is that legal?

Reply:
I see. there is not Apache 2.0 header. But still what about getting code from `SPDX-License-Identifier: GPL-2.0-only` based files?

Reply:
i think this is not the point of our discussion, the definition of 80211 is standard protocol, and the definition of linux is more comprehensive

Reply:
Yes. I'm fine to keep the original style. Just want to make sure that we are not hitting any legal issues so the whole project is not getting compromised.

Reply:
I have completed your change request, please review again

Reply:
@jerpelea could you please take a look?

Reply:
@pkarashchenko ieee80211.h exist for a long time, this PR doesn't add any new copyright issue. If this is a real problem, we need find an alternative solution, holding this PR doesn't change the truth.

Reply:
If initially this header was created from the same source, then I do not have any objection. Just want to be sure that we will not compromise Apache licence by bringing the the GPLv2 code

Reply:
Can we add people that can bring more light to this PR? Maybe request to start an e-mail discussion?

Reply:
Resolving this conversation since @jerpelea approved the change. Let's not wait anymore. Anyway I do not have enough competence in legal questions.

Reply:
@pkarashchenko please start a mailing list thread.